src/cargo/sources/git/utils.rs

```diff
@@ -732,6 +745,24 @@ pub fn fetch(
     })
 }
 
+fn fetch_with_cli(
+    repo: &mut git2::Repository,
+    url: &Url,
+    refspec: &str,
+    config: &Config,
+) -> CargoResult<()> {
+    let mut cmd = process("git");
```

Comment:
Any reason (either way) to use cargo's `process` API over the stdlib's `Command` API (like used by the `git gc` below)?

Reply:
Ah it mostly just had to do with how easy it was to deal with the result. The `git gc` piece is basically ignoring the result of the process no matter what happens, while here we want to report any failure of `git` and Cargo's already got good error handling and UI for all that.